apps/rebar/src/rebar_prv_manifest.erl

```diff
@@ -184,3 +184,17 @@ is_json_available() ->
         {error, _} ->
             false
     end.
+
+extra_src_dirs(Root, RebarOpts) ->
+    Extra = rebar_dir:extra_src_dirs(RebarOpts),
+    [Dir || E <- Extra, Dir <- expand(Root, E, rebar_dir:recursive(RebarOpts, E))].
+
+expand(_Root, Dir, false) ->
+    [Dir];
+expand(Root, Dir, true) ->
+    SubDirs = [
+        string:prefix(F, Root ++ "/") ||
```

Comment:
you might want to use [`rebar_file_utils:path_from_ancestor(File, ParentDir)`](https://github.com/erlang/rebar3/blob/7934da94f9d85febb1c2f27a86729665a30b4c8c/apps/rebar/src/rebar_file_utils.erl#L454) which will also handle cases like any stray relative path, period, but also drive letters or directory separators in windows paths.